### PR TITLE
Feat/dupe detection confirmation

### PR DESCRIPTION
**Pull request by kubabutkiewicz**
### Fixed Issues

$ https://github.com/Expensify/App/issues/39810

### Tests

1. Create some expanses which are duplicates
2. Go to one of expenses which are duplicates and click Review duplicates button
3. Next click on one of the Keep this one button
4. Now if there are some discrepancies between expenses you should be navigated to screen to review those discrepancies
5. When there are no discrepancies you will be navigated to confirm screen.
6. Verify that fields on confirmation screen are disabled.
7. Verify that fields which you selected during review are the same which are on confirmation
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Create some expanses which are duplicates
2. Go to one of expenses which are duplicates and click Review duplicates button
3. Next click on one of the Keep this one button
4. Now if there are some discrepancies between expenses you should be navigated to screen to review those discrepancies
5. When there are no discrepancies you will be navigated to confirm screen.
6. Verify that fields on confirmation screen are disabled.
7. Verify that fields which you selected during review are the same which are on confirmation
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
android

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>
ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
web

</details>

<details>
<summary>MacOS: Desktop</summary>
desktop

</details>

### CODE REVIEW

**Comment by JmillsExpensify** [comment]
@kubabutkiewicz when you're back on Monday, please make this PR your top focus. We're aiming to get this final PR for dupe detection merged by early next week!

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:4187]:
```suggestion
        confirmDetails: 'Confirma los detalles que conserva',
        confirmDuplicatesInfo: 'Los duplicados que no conserve se guardarán para que el usuario los elimine',
        hold: 'Bloqueado',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx
Comment (kubabutkiewicz) [reply src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx:63]: @pecanoro I remember that we were talking about that on the slack, can we add it to doc?

Comment (kubabutkiewicz) [reply src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx:63]: here: https://callstack-hq.slack.com/archives/C036QM0SLJK/p1715717883275529?thread_ts=1712851453.066999&cid=C036QM0SLJK

Comment (kubabutkiewicz) [reply src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx:63]: here: https://callstack-hq.slack.com/archives/C036QM0SLJK/p1715717883275529?thread_ts=1712851453.066999&cid=C036QM0SLJK

File: src/libs/actions/IOU.ts
Comment (kubabutkiewicz) [reply src/libs/actions/IOU.ts:7182]: this was the original solution but its produces TS error

File: src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx
Comment (pecanoro) [reply src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx:63]: Yeah, sounds good

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by kubabutkiewicz** [comment]
@parasharrajat this is ready to be check again

**Comment by pecanoro** [comment]
Is it ready for another review?

**Comment by JmillsExpensify** [comment]
What account are you using?

**Comment by JmillsExpensify** [comment]
Please reach out to me in Slack so that we can resolve your issue live. I'm hopeful that we can get this merged today. We only have one more day to get this in the deploy cycle ahead of our deadline.

**Comment by pecanoro** [comment]
> I am seeing Restricted add payment method page while making new requests. I currently have active free trial. Is there a way to bypass this? @pecanoro

Do you see the same in oldDot? The error message there might be clearer to see if there is an owed account or something

**Comment by pecanoro** [comment]
Also if you let me know the account, I can check the free trial date on my end and update it

**Comment by pecanoro** [comment]
I am also going to test the PR in the meantime

**Comment by pecanoro** [comment]
While testing, I noticed we still have the arrows/carrots on the confirmation page. Since they cannot be modified, we should remove them.

image

**Comment by pecanoro** [comment]
> While I agree, mock up does have arrows.

Ah yes, we were talking about it, I think originally we thought about being able to modify each field separately but I don't think we have time for that right now so we can handle that later as polish if we decide to add that functionality.

**Comment by pecanoro** [comment]
@JmillsExpensify noticed it is possible to add reactions to this screen. Can we remove that?

https://files.slack.com/files-pri/T047TPA624F-F07CZ355YNQ/cleanshot_2024-07-17_at_12.25.21_2x.png

**Comment by pecanoro** [comment]
Forget my previous comment, we can do that in a follow-up PR

**Comment by pecanoro** [comment]
> I reproduced this issue https://github.com/Expensify/App/pull/42571#issuecomment-2233853607 a few times now. @pecanoro Should this be fixed before we merge this PR?

Ah yes, that seems like an important one to fix

**Comment by kubabutkiewicz** [comment]
Yes, I will check it first thing in the morning

**Comment by kubabutkiewicz** [comment]
Also about arrows, I left them because it was like that in the design doc last time I check, should I remove them from a confirm page?

**Comment by kubabutkiewicz** [comment]
Just fyi I noticed that there is a problem with calculating tax amount but its also happening on main.
image

**Comment by kubabutkiewicz** [comment]
@parasharrajat Are you able to send to me a receipt which you used, when the bug with not found page occurred?

**Comment by JmillsExpensify** [comment]
@kubabutkiewicz no need to remove the arrows, since as you note, they are in the design doc.

**Comment by kubabutkiewicz** [comment]
Hey @parasharrajat, I pushed a fix for this redirect to not found page

**Comment by JmillsExpensify** [comment]
@parasharrajat @pecanoro Can you make reviewing this issue a priority today? We need to find a path for getting this merged in order to make our deadline.

**Comment by pecanoro** [comment]
Yes, I am doing another test run right now!

**Comment by pecanoro** [comment]
@parasharrajat How is it going?

**Comment by pecanoro** [comment]
The only bug I found is that after resolving/merging the duplicates, the report doesn't have the right total, it just shows as $0:

image

**Comment by kubabutkiewicz** [comment]
@pecanoro this seems to be a backend issue, I check the response for `OpenReport` command after resolving duplicates and there is something like that 
image

**Comment by pecanoro** [comment]
Oh wow, that is super odd indeed, I will create an issue to fix it internally then, thank you for looking into it! Waiting on @parasharrajat then to finish the whole review

**Comment by kubabutkiewicz** [comment]
@pecanoro I recheck what is sent to backend from frontend and I noticed that when there is no receipt amount in paramters for `MergeTransactions` is sent as 0 and that cause a problem pushing a fix for that

**Comment by JmillsExpensify** [comment]
Thanks for the quick work and responsiveness!

**Comment by pecanoro** [comment]
@parasharrajat Please prioritize the PR when you get online next! 🙏

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:4479]:
```suggestion
        confirmDetails: 'Confirma los detalles que conservas',
        confirmDuplicatesInfo: 'Los duplicados que no conserves se guardarán para que el usuario los elimine',
```

File: src/ROUTES.ts
Comment (pecanoro) [inline src/ROUTES.ts:949]: I don't want to block on this, but once we do any other polish issues, we should remove this extra line

```suggestion
```

File: src/libs/actions/IOU.ts
Comment (pecanoro) [inline src/libs/actions/IOU.ts:7358]:
```suggestion
/** Merge several transactions into one by updating the fields of the one we want to keep and deleting the rest */
function mergeDuplicates(params: TransactionMergeParams) {
```

File: src/libs/actions/IOU.ts
Comment (pecanoro) [inline src/libs/actions/IOU.ts:7418]:
```suggestion
```

File: src/libs/actions/IOU.ts
Comment (pecanoro) [inline src/libs/actions/IOU.ts:7422]:
```suggestion
```

File: src/pages/TransactionDuplicate/ReviewBillable.tsx
Comment (pecanoro) [inline src/pages/TransactionDuplicate/ReviewBillable.tsx:41]: Nothing to address, today I just learned that you can do this for the name of the component 😄

File: src/pages/TransactionDuplicate/ReviewTaxCode.tsx
Comment (pecanoro) [inline src/pages/TransactionDuplicate/ReviewTaxCode.tsx:50]: This should say what it does not what it handles, but just ignore it for now since it's too many changes 😄

**Review by pecanoro — DISMISSED** [review]
The comments that I left are pretty minor and can be addressed at some other PR

**Comment by pecanoro** [comment]
@kubabutkiewicz If you get online and this PR is not merged, feel free to commit my suggestions. If it's merged, no worries; we can handle them in some other Polish PR since we need to fix some minor stuff.

**Comment by JmillsExpensify** [comment]
@parasharrajat Just confirming that you are still reviewing/testing this PR?

**Comment by kubabutkiewicz** [comment]
@pecanoro I applied your suggestions

**Comment by kubabutkiewicz** [comment]
Trying to reproduce that but no luck yet

**Comment by kubabutkiewicz** [comment]
Checking it... but this seems to be a backend issue because I see in the response from `OpenReport` I see that there is a currency taken from a Workspace but I see I am sending a currency from a original transaction, but double checking

**Comment by kubabutkiewicz** [comment]
SCR-20240719-lcgl

**Comment by kubabutkiewicz** [comment]
image

**Comment by JmillsExpensify** [comment]
Nice, @pecanoro we're getting really close. QA only has 1% to go. Can we get this merged in the next hour?

**Review by pecanoro — APPROVED** [review]

**Comment by pecanoro** [comment]
I am merging so it goes into the deploy but just a note that I didn't retest the PR after the most recent commits. Hopefully it doesn't cause any issues, I will test on staging later today.

**Comment by pecanoro** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by pecanoro** [comment]
Also as a note, due to the urgency, I am copying and pasting the checklist but I didn't go through each item one by one

**Comment by pecanoro** [comment]
@parasharrajat It would have been great to have had that the checklist completed either way even though you found that small bug so it was ready to be merged this morning

**Comment by jayeshmangwani** [comment]
Coming from this issue https://github.com/Expensify/App/issues/45814, It seems we missed disabling the ReceiptEmptyState on confirmation screen



https://github.com/Expensify/App/blob/864c69f61e03669c20f2e0fcddd99f0291fb2240/src/components/ReportActionItem/MoneyRequestView.tsx#L477-L478

File: src/pages/TransactionDuplicate/Confirmation.tsx
Comment (rojiphil) [inline src/pages/TransactionDuplicate/Confirmation.tsx:76]: `MoneyRequestView` should have known that the view is displayed for dupe detection as the display of transaction receipt image depends on money request report id but the transaction itself may not belong to the same report. This caused the transaction receipt image to shown a not found image in #50107

File: src/libs/TransactionUtils.ts
Comment (alitoshmatov) [inline src/libs/TransactionUtils.ts:939]: Omitting other `reviewDuplicateTransaction.comment` properties caused an issue where if default tax is selected, `Default` label is not shown next to tax name

File: src/pages/TransactionDuplicate/Confirmation.tsx
Comment (jjcoffee) [inline src/pages/TransactionDuplicate/Confirmation.tsx:42]: Coming from https://github.com/Expensify/App/issues/58497. The resolve duplicates screen was still in the stack when tapping the back button after confirming duplicates. We should've used `dismissModal` before navigating here to resolve this.

**Pull request merged**
